data-prepper-plugins/drop-events-processor/src/main/java/com/amazon/dataprepper/plugins/processor/drop/HandleFailedEventsOption.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.drop;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+enum HandleFailedEventsOption {
+    DROP("drop", true, false),
+    DROP_SILENTLY("drop_silently", true, true),
+    SKIP("skip", false, false),
+    SKIP_SILENTLY("skip_silently", false, true);
+
+    private static final Logger LOG = LoggerFactory.getLogger(HandleFailedEventsOption.class);
+    private static final Map<String, HandleFailedEventsOption> OPTIONS_MAP = Arrays.stream(HandleFailedEventsOption.values())
+            .collect(Collectors.toMap(
+                    value -> value.option,
+                    value -> value
+            ));
+
+    private final String option;
+    private final boolean isDropEventOption;
+    private final boolean isLogRequired;
+
+    HandleFailedEventsOption(final String option, final boolean isDropEventOption, final boolean isLogRequired) {
+        this.option = option.toLowerCase();
+        this.isDropEventOption = isDropEventOption;
+        this.isLogRequired = isLogRequired;
+    }
+
+    public boolean isDropEventOption(final Event event, final Throwable cause) {
+        if (isLogRequired) {
+            LOG.warn("An exception occurred while processing when expression for event {}", event, cause);
```

Comment:
I think we should probably pass this logger in. It seems like the logging should come from `DropProcessor` instead of this class. I'd expect that is clearer when viewing logs.